src/javascript/app_2/Stores/Modules/Trading/Helpers/contract_type.js

```diff
@@ -209,24 +209,46 @@ const ContractType = (() => {
     const hours   = [...Array(24).keys()].map((a)=>`0${a}`.slice(-2));
     const minutes = [...Array(12).keys()].map((a)=>`0${a*5}`.slice(-2));
 
-    const getStartTime = (sessions, start_date, start_time) => {
-        let [ hour, minute ] = start_time.split(':');
-        const start_moment = moment.unix(start_date).utc().hour(hour).minute(minute);
-        if (sessions && !isSessionAvailable(sessions, start_moment)) {
-            hour   = hours.find(h => isSessionAvailable(sessions, start_moment.hour(h))) || hour;
-            minute = minutes.find(m => isSessionAvailable(sessions, start_moment.minute(m))) || minute;
+    const getValidTime = (sessions, compare_moment, start_moment) => {
+        if (sessions && !isSessionAvailable(sessions, compare_moment)) {
+            // first see if changing the minute brings it to the right session
+            compare_moment.minute(minutes.find(m => isSessionAvailable(sessions, compare_moment.minute(m))) || compare_moment.format('mm'));
+            // if not, also change the hour
+            if (!isSessionAvailable(sessions, compare_moment)) {
+                compare_moment.minute(0);
+                compare_moment.hour(hours.find(h => isSessionAvailable(sessions, compare_moment.hour(h), start_moment, true)) || compare_moment.format('HH'));
+                compare_moment.minute(minutes.find(m => isSessionAvailable(sessions, compare_moment.minute(m))) || compare_moment.format('mm'));
+            }
         }
-        return { start_time: `${hour}:${minute}` };
+        return compare_moment.format('HH:mm');
+    };
+
+    const buildMoment = (date, time) => {
+        const [ hour, minute ] = time.split(':');
+        return moment.utc(isNaN(date) ? date : +date * 1000).hour(hour).minute(minute);
```

Comment:
perhaps `moment.utc(isNaN(date) ? undefined : ...` ?

Reply:
no, in case where it's not a number it's a string format of date, so we want to change that to moment